react/GalleryLayout.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ComponentType, useContext, useMemo } from 'react'
+import classNames from 'classnames'
+
+import { ProductListContext } from 'vtex.product-list-context'
+import { Spinner } from 'vtex.styleguide'
+import { useCssHandles } from 'vtex.css-handles'
+import {
+  useResponsiveValue,
+  MaybeResponsiveInput,
+} from 'vtex.responsive-values'
+import { useRuntime } from 'vtex.render-runtime'
+import { useSearchPageState } from 'vtex.search-page-context/SearchPageContext'
+
+import GalleryLayoutRow from './components/GalleryLayoutRow'
+import SettingsContext from './components/SettingsContext'
+import ProductListEventCaller from './utils/ProductListEventCaller'
+
+export interface LayoutDescription {
+  name: string
+  component: string
+  itemsPerRow: MaybeResponsiveInput<number>
+}
+
+export type Slots = Record<string, ComponentType>
+
+export interface GalleryLayoutProps {
+  layouts: LayoutDescription[]
+  lazyItemsRemaining: number
+  products: any[]
+  showingFacets: boolean
+  summary: any
+  slots: Slots
+}
+
+export type GalleryLayoutPropsWithSlots = Omit<GalleryLayoutProps, 'slots'> &
+  Slots
```

Comment:
I would add a comment here, this looks a bit confusing at first sight, 'cause it looks like you're removing `slots` and adding it back again 😂

Reply:
hahahahah agreed!